lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -207,17 +205,21 @@ def size_allocate(self, widget, allocation):
         self.draw_idle()
 
     def _get_key(self, event):
+        unikey = chr(Gdk.keyval_to_unicode(event.keyval))
         key = cbook._unikey_or_keysym_to_mplkey(
-            chr(Gdk.keyval_to_unicode(event.keyval)),
+            unikey,
             Gdk.keyval_name(event.keyval))
         modifiers = [
-                     (Gdk.ModifierType.MOD4_MASK, 'super'),
-                     (Gdk.ModifierType.MOD1_MASK, 'alt'),
-                     (Gdk.ModifierType.CONTROL_MASK, 'ctrl'),
-                    ]
+            (Gdk.ModifierType.CONTROL_MASK, 'ctrl'),
+            (Gdk.ModifierType.MOD1_MASK, 'alt'),
+            (Gdk.ModifierType.SHIFT_MASK, 'shift'),
+            (Gdk.ModifierType.MOD4_MASK, 'super'),
+        ]
         for key_mask, prefix in modifiers:
             if event.state & key_mask:
-                key = '{0}+{1}'.format(prefix, key)
+                if not (prefix == 'shift' and unikey.isprintable()):
```

Comment:
in the tk backend this logic is `... and unikey`.  Is there a reason that gtk is different?

Reply:
There is. On `cbook._unikey_or_keysym_to_mplkey()` there is a comment that says:

> \# For non-printable characters, gtk3 passes "\0" whereas tk passes an "".

This return difference leads us to have different logic.